src/js_user.hpp

```diff
@@ -146,6 +148,15 @@ void UserClass<T>::get_state(ContextType ctx, ObjectType object, ReturnValue &re
     }
 }
 
+template<typename T>
+void UserClass<T>::get_custom_data(ContextType ctx, ObjectType object, ReturnValue &return_value) {
+    auto custom_data = get_internal<T, UserClass<T>>(object)->get()->custom_data();
+    if (!custom_data)
+        return_value.set_null();
+
+    return return_value.set(js::Value<T>::from_bson(ctx, *custom_data));
```

Comment:
Why is `return` needed here?

Reply:
It isn't. I must have typed it out of habit when typing the line to return something 🙂

Reply:
But a `return` *is* needed above.  😄 

It may be a good general convention to just always do `return return_value.set_foo(...);` even if the `return` is redundant. Then the js return also acts as a c++ return, reducing some of the impedance mismatch.